### ISSUE TITLE
1.22.2 does not build because add_backend_libs invoked with incorrect arguments

### ISSUE THREAD

**Issue by icebarf** [kcat/openal-soft#829]
Hi, I'm trying to build SFML/2.5.1 at my machine using Conan. The recipe for SFML on conan center currently lists openal/1.22.2 as a dependency.

The CMake output on the terminal is as follows:
```sh
-------- Building package openal/1.22.2 from source (20 of 21) --------
openal/1.22.2: Package openal/1.22.2:eae6bf279947d03ccadc33f9a9984209c88fad98
openal/1.22.2: Copying sources to build folder
openal/1.22.2: Building your package in /home/ice/.conan2/p/t/openaf91a2f9658a09/b
openal/1.22.2: Calling generate()
openal/1.22.2: Generators folder: /home/ice/.conan2/p/t/openaf91a2f9658a09/b/build/Release/generators
openal/1.22.2: CMakeToolchain generated: conan_toolchain.cmake
openal/1.22.2: CMakeToolchain generated: CMakePresets.json
openal/1.22.2: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
openal/1.22.2: Generating aggregated env files
openal/1.22.2: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
openal/1.22.2: Calling build()
openal/1.22.2: Running CMake.configure()
openal/1.22.2: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/ice/.conan2/p/t/openaf91a2f9658a09/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/ice/.conan2/p/t/openaf91a2f9658a09/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/ice/.conan2/p/t/openaf91a2f9658a09/b/src"
-- Using Conan toolchain: /home/ice/.conan2/p/t/openaf91a2f9658a09/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Looking for posix_memalign
-- Looking for posix_memalign - found
-- Performing Test HAVE___RESTRICT
-- Performing Test HAVE___RESTRICT - Success
-- Performing Test HAVE_LIBATOMIC
-- Performing Test HAVE_LIBATOMIC - Success
-- Performing Test HAVE_FNO_MATH_ERRNO
-- Performing Test HAVE_FNO_MATH_ERRNO - Success
-- Performing Test HAVE_GCC_PROTECTED_VISIBILITY
-- Performing Test HAVE_GCC_PROTECTED_VISIBILITY - Success
-- Performing Test HAVE_VISIBILITY_HIDDEN_SWITCH
-- Performing Test HAVE_VISIBILITY_HIDDEN_SWITCH - Success
-- Performing Test HAVE_MSSE2_SWITCH
-- Performing Test HAVE_MSSE2_SWITCH - Success
-- Looking for xmmintrin.h
-- Looking for xmmintrin.h - found
-- Looking for emmintrin.h
-- Looking for emmintrin.h - found
-- Looking for pmmintrin.h
-- Looking for pmmintrin.h - found
-- Looking for smmintrin.h
-- Looking for smmintrin.h - found
-- Looking for arm_neon.h
-- Looking for arm_neon.h - not found
-- Performing Test HAVE_SSE_INTRINSICS
-- Performing Test HAVE_SSE_INTRINSICS - Success
-- Looking for malloc.h
-- Looking for malloc.h - found
-- Looking for cpuid.h
-- Looking for cpuid.h - found
-- Looking for intrin.h
-- Looking for intrin.h - not found
-- Looking for guiddef.h
-- Looking for guiddef.h - not found
-- Looking for initguid.h
-- Looking for initguid.h - not found
-- Looking for pow in m
-- Looking for pow in m - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for dlfcn.h
-- Looking for dlfcn.h - found
-- Looking for dlopen in dl
-- Looking for dlopen in dl - found
-- Performing Test HAVE_GCC_GET_CPUID
-- Performing Test HAVE_GCC_GET_CPUID - Success
-- Looking for posix_memalign
-- Looking for posix_memalign - found
-- Looking for _aligned_malloc
-- Looking for _aligned_malloc - not found
-- Looking for proc_pidpath
-- Looking for proc_pidpath - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test HAVE_PTHREAD
-- Performing Test HAVE_PTHREAD - Success
-- Looking for pthread_setschedparam
-- Looking for pthread_setschedparam - found
-- Looking for include files pthread.h, pthread_np.h
-- Looking for include files pthread.h, pthread_np.h - not found
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - not found
-- Looking for pthread_set_name_np
-- Looking for pthread_set_name_np - not found
-- Looking for getopt
-- Looking for getopt - found
-- Found ALSA: /home/ice/.conan2/p/libal0651245d7b098/p/lib/libasound.a (found version "1.2.7.2") 
-- Found OSS: /usr/include  
-- Checking for module 'libpipewire-0.3>=0.3.23'
--   Found libpipewire-0.3, version 0.3.66
-- Could NOT find AudioIO (missing: AUDIOIO_INCLUDE_DIR) 
-- Found PortAudio: /usr/lib/libportaudio.so  
CMake Error at CMakeLists.txt:1054 (add_backend_libs):
  add_backend_libs Macro invoked with incorrect arguments for macro named:
  ADD_BACKEND_LIBS


-- Found JACK: /usr/lib/libjack.so  
-- Could NOT find OpenSL (missing: OPENSL_LIBRARY OPENSL_INCLUDE_DIR OPENSL_ANDROID_INCLUDE_DIR) 
-- Found Git: /usr/bin/git (found version "2.39.2") 
-- 
-- Building OpenAL with support for the following backends:
--     ALSA, OSS, PipeWire, PortAudio, PulseAudio, JACK, WaveFile, Null
-- 
-- Building with support for CPU extensions:
--     Default, SSE, SSE2, SSE3, SSE4.1
-- 
-- Embedding HRTF datasets
-- 
-- Installing library and headers
-- Installing sample configuration
-- Installing HRTF data files
-- Installing AmbDec presets
-- 
-- Configuring incomplete, errors occurred!
See also "/home/ice/.conan2/p/t/openaf91a2f9658a09/b/build/Release/CMakeFiles/CMakeOutput.log".
See also "/home/ice/.conan2/p/t/openaf91a2f9658a09/b/build/Release/CMakeFiles/CMakeError.log".
```

I have attached above log files as well.
[CMakeError.log](https://github.com/kcat/openal-soft/files/10860282/CMakeError.log)
[CMakeOutput.log](https://github.com/kcat/openal-soft/files/10860283/CMakeOutput.log)

As it can be seen that CMake indicates the issue as follows:

```sh
CMake Error at CMakeLists.txt:1054 (add_backend_libs):
  add_backend_libs Macro invoked with incorrect arguments for macro named:
  ADD_BACKEND_LIBS
```

Here's a link to the relevant part of [CMakeLists.txt](https://github.com/kcat/openal-soft/blob/dc83d99c95a42c960150ddeee06c124134b52208/CMakeLists.txt#L1054)

If more information is needed, please specify. Thank you!

**Comment by kcat** [comment]
Looks like it's using a config module from PulseAudio, which doesn't set certain variables. That should be fixed with 1.23, or can be fixed for 1.22.2 by changing
```cmake
add_backend_libs(${PULSEAUDIO_LIBRARIES})
set(INC_PATHS ${INC_PATHS} ${PULSEAUDIO_INCLUDE_DIRS})
```
to
```cmake
add_backend_libs(${PULSEAUDIO_LIBRARY})
set(INC_PATHS ${INC_PATHS} ${PULSEAUDIO_INCLUDE_DIR})
```

**Comment by icebarf** [comment]
Thank you, I will try and let you know if it is fixed
